internal/rbd/nodeserver.go

```diff
@@ -718,8 +719,20 @@ func (ns *NodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 	}
 
 	diskMounter := &mount.SafeFormatAndMount{Interface: ns.mounter, Exec: utilexec.New()}
-	// TODO check size and return success or error
 	volumePath += "/" + volumeID
+
+	// check size and return success or error
+	var vStat syscall.Statfs_t
+	err = syscall.Statfs(volumePath, &vStat)
+	if err != nil {
+		return nil, status.Errorf(codes.Aborted, util.VolumePathStatfsFailedFmt, req.GetVolumePath())
+	}
+	volSize := util.RoundOffBytes(req.GetCapacityRange().GetRequiredBytes())
+	vPathSize := util.RoundOffBytes(vStat.Bsize * int64(vStat.Blocks))
+	if vPathSize == volSize {
```

Comment:
@agarwal-mudit  If both are equal , we dont have to error out, rather say the nodeExpandVolume was successful . Isnt it ?

Reply:
What if  the CSI driver prematurely issued the FS resize before krbd noticed that the volume was resized like what Jason said in https://bugzilla.redhat.com/show_bug.cgi?id=1868703#c10
In that case also both sizes would be same, shouldn't we fail and retry in such case or I am missing something.

Reply:
This is against the CSI spec. we return success as the CSI call should be idempotent.